src/algebra/group_power/lemmas.lean

```diff
@@ -132,6 +132,26 @@ by rw [zpow_bit0', zpow_bit0', neg_mul_neg]
 
 end division_monoid
 
+section pow_mul_action
+
+variables (M G) [monoid M] [group G]
+
+/-- Exponentiation by natural numbers as a `mul_action`. -/
+@[to_additive nsmul_mul_action "Multiplication by natural numbers as a `mul_action`."]
+def pow_mul_action : mul_action ℕ M :=
+{ smul := λ n g, g^n,
+  one_smul := pow_one,
+  mul_smul := λ m n g, pow_mul' g m n }
+
+/-- Exponentiation by integers as a `mul_action`. -/
+@[to_additive zsmul_mul_action "Multiplication by integers as a `mul_action`."]
+def zpow_mul_action : mul_action ℤ G :=
+{ smul := λ n g, g^n,
+  one_smul := zpow_one,
+  mul_smul := λ m n g, zpow_mul' g m n }
```

Comment:
I don't like the look of these defs; they feel like exposing a kind of `monoid.to_add_monoid` def that swaps `*` with `+`.

The additive version this generates is just the same as the instance that's embedded in `add_group G`.

I think the right thing to do is prove the general case with a `monoid_hom`. I'll push a fix.

Reply:
Hmm, I couldn't get my version to work out nicely. I think your original ungolfed version was fine, and I just misinterpreted what the the `to_additive` was doing

Reply:
Thank you very much for trying.

In that case, I'll remove these `pow_mul_action` `def`s and prove what I want by inlining them.

Reply:
(now done)

Reply:
Thanks!